File: car-rental-system-frontend/src/FavoriteComponent/favoriteService.js
import axios from "axios";
import { API_BASE_URL } from "../apiConfig";

class FavoriteService {
    addToFavorites(favoriteRequest) {
        return axios.post(API_BASE_URL + "/favorite/add", favoriteRequest);
    }

    getFavoritesByCustomer(customerId) {
        return axios.get(API_BASE_URL + "/favorite/fetch/user-wise?customerId=" + customerId);
    }

    checkIfFavorited(carId, customerId) {
        return axios.get(API_BASE_URL + "/favorite/check?carId=" + carId + "&customerId=" + customerId);
    }

    removeFromFavorites(carId, customerId) {
        return axios.delete(API_BASE_URL + "/favorite/remove?carId=" + carId + "&customerId=" + customerId);
    }
}

export default new FavoriteService();

File: car-rental-system-backend/src/main/java/com/carrentalsystem/service/FavoriteService.java
package com.carrentalsystem.service;

import java.util.List;

import com.carrentalsystem.dto.CreateFavoriteRequest;
import com.carrentalsystem.entity.Favorite;
import com.carrentalsystem.entity.Variant;

public interface FavoriteService {

    void addToFavorites(CreateFavoriteRequest request);

    List<Variant> getFavoritesByCustomer(int customerId);

    boolean checkIfFavorited(int carId, int customerId);

    void removeFromFavorites(int carId, int customerId);

}
